hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenKeysHelper.java

```diff
@@ -0,0 +1,233 @@
+        },
+
+        FEE_SCHEDULE_KEY {
+            @Override
+            public boolean isPresentInUpdate(TokenUpdateTransactionBody update) {
+                return update.hasFeeScheduleKey();
+            }
+
+            @Override
+            public boolean isPresentInitially(Token originalToken) {
+                return originalToken.hasFeeScheduleKey();
+            }
+
+            @Override
+            public void setOn(final Token.Builder builder, TokenUpdateTransactionBody update) {
+                builder.feeScheduleKey(getNewKeyValue(update.feeScheduleKey()));
+            }
+
+            @Override
+            ResponseCodeEnum tokenHasNoKeyStatus() {
+                return TOKEN_HAS_NO_FEE_SCHEDULE_KEY;
+            }
+        },
+
+        PAUSE_KEY {
+            @Override
+            public boolean isPresentInUpdate(TokenUpdateTransactionBody update) {
+                return update.hasPauseKey();
+            }
+
+            @Override
+            public boolean isPresentInitially(Token originalToken) {
+                return originalToken.hasPauseKey();
+            }
+
+            @Override
+            public void setOn(final Token.Builder builder, TokenUpdateTransactionBody update) {
+                builder.pauseKey(getNewKeyValue(update.pauseKey()));
+            }
+
+            @Override
+            ResponseCodeEnum tokenHasNoKeyStatus() {
+                return TOKEN_HAS_NO_PAUSE_KEY;
+            }
+        },
+
+        METADATA_KEY {
+            @Override
+            public boolean isPresentInUpdate(TokenUpdateTransactionBody update) {
+                return update.hasMetadataKey();
+            }
+
+            @Override
+            public boolean isPresentInitially(Token originalToken) {
+                return originalToken.hasMetadataKey();
+            }
+
+            @Override
+            public void setOn(final Token.Builder builder, TokenUpdateTransactionBody update) {
+                builder.metadataKey(getNewKeyValue(update.metadataKey()));
+            }
+
+            @Override
+            ResponseCodeEnum tokenHasNoKeyStatus() {
+                return TOKEN_HAS_NO_METADATA_KEY;
+            }
+        };
+
+        public abstract boolean isPresentInUpdate(TokenUpdateTransactionBody update);
+
+        public abstract boolean isPresentInitially(Token originalToken);
+
+        public abstract void setOn(final Token.Builder builder, TokenUpdateTransactionBody update);
+
+        abstract ResponseCodeEnum tokenHasNoKeyStatus();
+
+        private static Key getNewKeyValue(Key newKey) {
+            return isKeyRemoval(newKey) ? null : newKey;
+        }
+
+        public void updateKey(TokenUpdateTransactionBody update, Token originalToken, final Token.Builder builder) {
+            if (isPresentInUpdate(update)) {
+                validateTrue(isPresentInitially(originalToken), tokenHasNoKeyStatus());
+                setOn(builder, update);
+            }
+        }
```

Comment:
You could also add `isValid` method here. So, we can validate all keys

Reply:
@Neeharika-Sompalli 
That was our initial idea but method `validateTokenKeys()` is called from both `TokenCreateValidator` and `TokenUpdateValidator` which means we have 2 different operation structures and cannot unify them with the same approach..